oauth2/src/main/java/com/linecorp/armeria/common/auth/oauth2/GrantedOAuth2AccessToken.java

```diff
@@ -0,0 +1,407 @@
+     * authorization grant as described at
+     * <a href="http://tools.ietf.org/html/rfc6749#section-6">[RFC6749], Section 6</a>.
+     */
+    @Nullable
+    private final String refreshToken;
+
+    /**
+     * {@code scope} Access Token response field,
+     * OPTIONAL, if identical to the scope requested by the client; otherwise, REQUIRED. The scope of
+     * the access token as described at
+     * <a href="http://tools.ietf.org/html/rfc6749#section-3.3">[RFC6749], Section 3.3</a>.
+     * A list of space-delimited, case-sensitive scope strings. The strings are defined by the authorization
+     * server.
+     * The authorization server MAY fully or partially ignore the scope requested by the client, based
+     * on the authorization server policy or the resource owner's instructions.  If the issued access
+     * token scope is different from the one requested by the client, the authorization server MUST
+     * include the "scope" response parameter to inform the client of the actual scope granted.
+     * If the client omits the scope parameter when requesting authorization, the authorization server
+     * MUST either process the request using a pre-defined default value or fail the request
+     * indicating an invalid scope.
+     */
+    @Nullable
+    private final String scope;
+
+    /**
+     * A {@link Set} of case-sensitive scope strings. The strings are defined by the authorization
+     * server.
+     */
+    private final Set<String> scopeSet;
+
+    /**
+     * A {@link Map} of extra system-specific token parameters included with Access Token response,
+     * OPTIONAL.
+     */
+    private final Map<String, String> extras;
+
+    private final Instant issuedAt;
+
+    @Nullable
+    private String rawResponse;
+
+    @Nullable
+    private transient Instant expiresAt;
+
+    @Nullable
+    private transient String authorization;
+
+    @Nullable
+    private transient String toString;
+
+    GrantedOAuth2AccessToken(String accessToken, @Nullable String tokenType,
+                             Instant issuedAt, @Nullable Duration expiresIn,
+                             @Nullable String refreshToken, @Nullable List<String> scopeList,
+                             @Nullable ImmutableMap<String, String> extras,
+                             @Nullable String rawResponse) {
+        // token fields
+        this.accessToken = requireNonNull(accessToken, ACCESS_TOKEN);
+        this.tokenType = tokenType;
+        this.expiresIn = expiresIn;
+        this.issuedAt = requireNonNull(issuedAt, ISSUED_AT);
+        this.refreshToken = refreshToken;
+        scope = toScopeString(scopeList);
+        scopeSet = (scopeList == null) ? ImmutableSet.of() : ImmutableSet.copyOf(scopeList);
+        this.extras = (extras == null) ? ImmutableMap.of() : extras;
+        // raw response
+        this.rawResponse = rawResponse;
+    }
+
+    /**
+     * {@code access_token} Access Token response field,
```

Comment:
Global comment: Could use `{@value ...}` expression, e.g. `{@value #ACCESS_TOKEN}`.

Reply:
Somehow it does not work well with static imports

Reply:
Does `{@value ClassName#FIELD_NAME}` work, then?

Reply:
yes. fixed